### ISSUE TITLE
ImportOrder: make static imports ordering as in Eclipse

### ISSUE THREAD

**Issue by romani** [checkstyle/checkstyle#3101]
base on https://bugs.eclipse.org/bugs/show_bug.cgi?id=473629

```
/var/tmp $ java -jar checkstyle-6.17-SNAPSHOT-all.jar -c Test.xml TestClass.java 
Starting audit...
[ERROR] /var/tmp/TestClass.java:4: Wrong order for 'HttpHeaders.Names.DATE' import. [ImportOrder]
Audit done.
Checkstyle ends with 1 errors.

/var/tmp $ cat TestClass.java 
import static HttpConstants.COLON;
import static HttpHeaders.addHeader;
import static HttpHeaders.setHeader;
import static HttpHeaders.Names.DATE;

public class TestClass {

}

/var/tmp $ cat Test.xml 
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name = "Checker">
    <module name="TreeWalker">
      <module name="ImportOrder">
        <property name="groups" value="/^javax?\./,org"/>
        <property name="ordered" value="true"/>
        <property name="separated" value="true"/>
        <property name="option" value="top"/>
        <property name="sortStaticImportsAlphabetically" value="true"/>
      </module>
    </module>
</module>
```

Expected, no violations for Eclipse users.

investigate sorting behavior for static imports in Idea and NetBeans.

If the same in all 3 IDEs - just follow them
If different - we need new properties with explanations.

**Comment by rnveach** [comment]
NetBeans default, imports ordered by Class name, ignoring static/nonstatic. Looks like classes come before methods?
````
import static test3101.HttpConstants.COLON;
import test3101.HttpHeaders;
import static test3101.HttpHeaders.Names.DATE;
import static test3101.HttpHeaders.addHeader;
import static test3101.HttpHeaders.setHeader;
````

IntelliJ default, non-static first, then static.
````
import test3101.HttpHeaders;

import static test3101.HttpConstants.COLON;
import static test3101.HttpHeaders.Names.DATE;
import static test3101.HttpHeaders.addHeader;
import static test3101.HttpHeaders.setHeader;
````

Eclipse default, static first, looks complete alphabetical order afterwards.
````
import static test3101.HttpConstants.COLON;
import static test3101.HttpHeaders.addHeader;
import static test3101.HttpHeaders.setHeader;
import static test3101.HttpHeaders.Names.DATE;
import test3101.HttpHeaders;
````

**Comment by romani** [comment]
@rnveach ,
thanks a lot for quick reaction on this issue.
Main problem is that Eclipse use non ASCII (http://www.asciitable.com/) approach for sorting (described in Eclipse issue). NetBeans and Idea - ASCII .

We need new option for this Check - "useContainerOrderingForStatic" = "true/false". Default is false (as 2 of 3 IDEs use this approach).

**Comment by MEZk** [comment]
@romani 
Should we take into account the value of 'caseSensitive' option when we compare static import containers ?

**Comment by romani** [comment]
> Should we take into account the value of 'caseSensitive' option when we compare static import containers?

yes

**Comment by romani** [comment]
Fix is merged
